python/lsst/analysis/drp/skyPlot.py

```diff
@@ -294,41 +329,78 @@ def skyPlot(self, catPlot, plotInfo, sumStats):
+                ra = corners[0][0].asDegrees()
+                dec = corners[0][1].asDegrees()
+                xy = (ra, dec)
+                width = corners[2][0].asDegrees() - ra
+                height = corners[2][1].asDegrees() - dec
+                patches.append(Rectangle(xy, width, height, alpha=0.3))
+                ras = [ra.asDegrees() for (ra, dec) in corners]
+                decs = [dec.asDegrees() for (ra, dec) in corners]
+                ax.plot(ras + [ras[0]], decs + [decs[0]], "k", lw=0.5)
+                cenX = ra + width / 2
+                cenY = dec + height / 2
+                if dataId == "tract":
+                    minRa = np.min(ras)
+                    minDec = np.min(decs)
+                    maxRa = np.max(ras)
+                    maxDec = np.max(decs)
+                if dataId != "tract":
+                    ax.annotate(dataId, (cenX, cenY), color="k", fontsize=5, ha="center", va="center",
+                                path_effects=[pathEffects.withStroke(linewidth=2, foreground="w")])
 
         for (i, (xs, ys, colorVals, cmap, label)) in enumerate(toPlotList):
+            if "tract" not in sumStats.keys() or not self.config.plotOutlines:
+                minRa = np.min(xs)
+                maxRa = np.max(xs)
+                minDec = np.min(ys)
+                maxDec = np.max(ys)
             med = np.median(colorVals)
             mad = sigmaMad(colorVals)
-            vmin = med - 3*mad
-            vmax = med + 3*mad
-            scatterOut = ax.scatter(xs, ys, c=colorVals, cmap=cmap, s=4.0, vmin=vmin, vmax=vmax)
+            vmin = med - 2*mad
+            vmax = med + 2*mad
+            if self.config.fixAroundZero:
+                scaleEnd = np.max([np.fabs(vmin), np.fabs(vmax)])
+                vmin = -1*scaleEnd
+                vmax = scaleEnd
+            nBins = 50
+            xBins = np.linspace(minRa, maxRa, nBins)
+            yBins = np.linspace(minDec, maxDec, nBins)
+            binnedStats, xEdges, yEdges, binNums = binned_statistic_2d(xs, ys, colorVals, statistic="median",
+                                                                       bins=(xBins, yBins))
+
+            if len(xs) > 10000:
+                s = 50000/(len(xs)**1.1)
+                plotOut = ax.imshow(binnedStats.T, cmap=cmap,
+                                    extent=[xEdges[0], xEdges[-1], yEdges[-1], yEdges[0]], vmin=vmin,
+                                    vmax=vmax)
+                # find the most extreme 15% of points, because the list
+                # is ordered by the distance from the median this is just
+                # the final 15% of points
+                extremes = int(np.floor((len(xs)/100))*85)
+                ax.scatter(xs[extremes:], ys[extremes:], c=colorVals[extremes:], s=s, cmap=cmap, vmin=vmin,
+                           vmax=vmax)
+
+            else:
+                plotOut = ax.scatter(xs, ys, c=colorVals, cmap=cmap, s=7, vmin=vmin, vmax=vmax)
```

Comment:
After much testing, I find this setup to work much better:
```
            if len(xs) > 5000:
                s = 500/(len(xs)**0.5)
                lw = (s**0.5) / 10
                plotOut = ax.imshow(binnedStats.T, cmap=cmap,
                                    extent=[xEdges[0], xEdges[-1], yEdges[-1], yEdges[0]], vmin=vmin,
                                    vmax=vmax)
                # find the most extreme 15% of points, because the list
                # is ordered by the distance from the median this is just
                # the final 15% of points
                extremes = int(np.floor((len(xs)/100))*85)
                ax.scatter(xs[extremes:], ys[extremes:], c=colorVals[extremes:], s=s, cmap=cmap,
                           vmin=vmin, vmax=vmax, edgecolor="white", linewidths=lw)

            else:
                plotOut = ax.scatter(xs, ys, c=colorVals, cmap=cmap, s=7, vmin=vmin, vmax=vmax,
                                     edgecolor="white", linewidths=0.2)
```

Reply:
It reduces the drop-off in size for large numbers of data points, as I found they were becoming too small to see for large volume plots. Furthermore, it adds a small white border around each point. I also reduced the threshold at which the transition from scatter-plot to hybrid scatter/histogram takes place from 10000 to 5000 points. For the record, I'm still not a huge fan of this hybrid plot, but, I realize that it had group consensus in discussions we've had before, so I'm happy to go with it here.